marimbabot_hardware/src/servo_interface_wifi.cpp

```diff
@@ -0,0 +1,231 @@
+        servo_state.name, &servo_state.position, &servo_state.velocity, &servo_state.effort);
+    joint_state_interface.registerHandle(servo_state_handle);
+
+    registerInterface(&joint_state_interface);
+
+    // Creating the servo position control interface + handle
+    hardware_interface::JointHandle position_servo_handle(
+        joint_state_interface.getHandle(servo_state.name), &servo_state.command);
+    position_joint_interface.registerHandle(position_servo_handle);
+
+    registerInterface(&position_joint_interface);
+
+    // Creating the joint saturation interface + handle
+    joint_limits_interface::getJointLimits(servo_state.name, node_handle, servo_limits);
+    joint_limits_interface::PositionJointSaturationHandle saturation_handle(
+        position_servo_handle, servo_limits);
+    position_joint_saturation_interface.registerHandle(saturation_handle);
+
+    registerInterface(&position_joint_saturation_interface);
+}
+
+void ServoInterface::initialize() {
+    // Keeping the current position of the servo
+    read();
+    servo_state.command = servo_state.position;
+    previous_command = servo_state.command;
+}
+
+bool ServoInterface::try_open_udp_port() {
+   
+        //Bind the Socket with the address information
+        bind(sockfd, (const struct sockaddr *)&servaddr, sizeof(servaddr)); 
+
+        ROS_INFO("Servo controller: Successfully opened port " );
+
+        len = sizeof(cliaddr);  //len is value/result
+        std::string response; 
+	    response = send_and_receive(ServoInterface::l_Sender);
+       
+
+}
+void ServoInterface::read() {
+    // Calculating time since last update
+    ros::Time current_time = ros::Time::now();
+    last_run_period = current_time - last_run_time;
+    last_run_time = current_time;
+
+    send_and_receive(ServoInterface::p_Sender);
+    
+}
+void ServoInterface::write(){
+    
+    position_joint_saturation_interface.enforceLimits(last_run_period);
+
+    if(previous_command == servo_state.command) {
+        ROS_INFO("Position = Position");
+        return;
+    }
+    ROS_DEBUG_STREAM("Servo controller: Sending command to arduino: " << servo_state.command);
+
+    // Remapping pwm value from radians input
+    int command_value = top_limit - (int) round((servo_state.command / radian_limit) * (top_limit - bottom_limit));
+    // Sending command to arduino
+    std::string s_Sender = "s ";
+    ROS_INFO_STREAM(command_value);
+    s_Sender += std::to_string(command_value);
+    s_Sender +=("\n");
+    char* write_pos_array = convert_to_char(s_Sender);
+    ROS_INFO_STREAM(s_Sender);
+  
+    send_and_receive(write_pos_array);
+    
+    
+}
+
+char* ServoInterface::send_and_receive(char * message){
+    memset(&buffer, 0, sizeof(buffer));
+     ROS_INFO_STREAM(message);
+    sendto(sockfd, (const char *)message, 
+    strlen(message),MSG_CONFIRM, (const struct sockaddr *) &servaddr,len);
+
+    received_message = recvfrom(sockfd, (char *)buffer, MAXLINE, 
```

Comment:
I'm just now thinking, if the message gets lost, you won't receive anything here and wait for some time.
It might be needed to handle responses asyncronously